smart-metastore/src/main/java/org/smartdata/metastore/dao/FileStateDao.java

```diff
@@ -60,10 +60,15 @@ public FileState getByPath(String path) {
         new FileStateRowMapper());
   }
 
-  public void deleteByPath(String path) {
+  public void deleteByPath(String path, boolean recursive) {
     JdbcTemplate jdbcTemplate = new JdbcTemplate(dataSource);
-    final String sql = "DELETE FROM " + TABLE_NAME + " WHERE path = ?";
-    jdbcTemplate.update(sql, path);
+    if (recursive) {
+      final String sql = "DELETE FROM " + TABLE_NAME + " WHERE path LIKE ?";
```

Comment:
Not correct. Remember our previous bug, i.e. delete file_1 results delete_file_1%.